src/ChunkedCSV.jl

```diff
@@ -109,19 +110,37 @@ struct ParserSettings
     end
 end
 
-_is_supported_type(::Type{T}) where {T} = false
-_is_supported_type(::Type{T}) where {T<:Union{Int,Float64,String,Char,Bool,DateTime,Date}} = true
-_is_supported_type(::Type{FixedDecimal{T,f}}) where {T<:Union{Int8,Int16,Int32,Int64,Int128,UInt8,UInt16,UInt32,UInt64,UInt128},f} = f <= 8
+_is_supported_type(::Type{T}) where {T} = Parsers.supportedtype(T)
+_is_supported_type(::Type{Nothing}) = true
+_is_supported_type(::Type{GuessDateTime}) = true
+function _is_supported_type(::Type{FixedDecimal{T,f}}) where {T,f}
+    # https://github.com/JuliaMath/FixedPointDecimals.jl/blob/1328b9a372d2285765a7255f154f09ffdd692508/src/FixedPointDecimals.jl#L83-L91
+    n = FixedPointDecimals.max_exp10(T)
+    return f >= 0 && (n < 0 || f <= n)
+end
 function validate_schema(types::Vector{DataType})
-    unsupported_types = filter(!_is_supported_type, types)
+    unsupported_types = unique!(filter(!_is_supported_type, types))
     if !isempty(unsupported_types)
-        err_msg = "Provided schema contains unsupported types: $(join(unique!(unsupported_types), ", "))."
-        any(T->(T <: FixedDecimal), unsupported_types) && (err_msg *= " Note: Currently, only decimals with less than 9 decimal places are supported.")
+        err_msg = "Provided schema contains unsupported types: $(join(unsupported_types, ", "))."
         throw(ArgumentError(err_msg))
     end
     return nothing
 end
 
+# Separate out the types that are not pre-compiled by the parser by default
+# and return them as a single Tuple of unique types which can be passed to
+# _parse_rows_forloop! to trigger recompilation.
+function _custom_types(schema::Vector{DataType})
+    custom_types = sort!(collect(keys(
+            IdDict{Type,Nothing}(
+                T => nothing for T in schema if isnothing(get(Enums._MAPPING, T, nothing))
+            ))),
```

Comment:
why do we construct a `IdDict` just to then convert to a `Vector`?
Also can the `isnothing` check just be `haskey`?

```suggestion
    custom_types = sort!(
            Type[T for T in schema if !haskey(Enums._MAPPING, T)];
```

(also why `Type` not `DataType`?)

Reply:
I need to remove duplicate `Type`s, so `IdDict{Type,Nothing}` serves like a `IdSet` for this purpose. I didn't notice any performance difference between `Type` and `DataType`  so I kept the more general one. `haskey` performed worse:
```julia
# haskey
julia> @btime ChunkedCSV._custom_types($([String,Int32,Int,Float32]))
  158.094 ns (5 allocations: 432 bytes)
Tuple{Int32, Float32}

# isnothing
julia> @btime ChunkedCSV._custom_types($([String,Int32,Int,Float32]))
  148.137 ns (3 allocations: 400 bytes)
Tuple{Int32, Float32}
```